gslib/utils/boto_util.py

```diff
@@ -499,6 +485,49 @@ def ResumableThreshold():
   return config.getint('GSUtil', 'resumable_threshold', 8 * ONE_MIB)
 
 
+def SetProxyInfo():
+  """Sets proxy info from boto and environment and converts to httplib2.ProxyInfo.
+
+  Args:
+    None.
+
+  Returns:
+    httplib2.ProxyInfo constructed from boto or environment variable string.
+  """
+  #Defining proxy_type based on httplib2 library, accounting for None entry too.
+  proxy_type_spec = {'socks4': 1, 'socks5': 2, 'http': 3, 'https': 3}
+
+  #proxy_type defaults to 'http (3)' for backwards compatibility
+  proxy_type = proxy_type_spec.get(config.get('Boto', 'proxy_type', None)) or 3
```

Comment:
I think it would be more clear to make this line: 
`proxy_type_spec.get(config.get('Boto', 'proxy_type', proxy_type_spec.https))`

Reply:
changed code to: 
`boto_proxy_val = config.get('Boto', 'proxy_type', None)
    
 #proxy_type defaults to 'http (3)' for backwards compatibility
 proxy_type = proxy_type_spec.get(boto_proxy_val) or proxy_type_spec['http']`